scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,351 @@
+    return FileInputStream(File(pathToPbFile)).use {
+      proto.newBuilderForType().mergeFrom(it)
+    }.build() as MavenDependencyList
+  }
+
+  private fun omitVersionAndReplaceColonsHyphensPeriods(artifactName: String): String {
+    val lastColonIndex = artifactName.lastIndexOf(':')
+    return artifactName.substring(0, lastColonIndex).replace('.', '_').replace(':', '_')
+      .replace('-', '_')
+  }
+
+  private fun genearateDependenciesListFromMavenInstall(
+    pathToMavenInstall: String,
+    bazelQueryDepsNames: List<String>
+  ): List<MavenListDependency> {
+    val mavenInstallJson = File(pathToMavenInstall)
+    val mavenInstallJsonText =
+      mavenInstallJson.inputStream().bufferedReader().use { it.readText() }
+    val moshi = Moshi.Builder().addLast(KotlinJsonAdapterFactory()).build()
+    val adapter = moshi.adapter(MavenListDependencyTree::class.java)
+    val dependencyTree = adapter.fromJson(mavenInstallJsonText)
+    val mavenInstallDependencyList = dependencyTree?.mavenListDependencies?.dependencyList
+    return mavenInstallDependencyList?.filter { dep ->
+      bazelQueryDepsNames.contains(omitVersionAndReplaceColonsHyphensPeriods(dep.coord))
+    } ?: listOf<MavenListDependency>()
+  }
+
+  private fun writeTextProto(
+    pathToTextProto: String,
+    mavenDependencyList: MavenDependencyList
+  ) {
+    File(pathToTextProto).outputStream().bufferedWriter().use { writer ->
+      TextFormat.printer().print(mavenDependencyList, writer)
+    }
+  }
+
+  private fun retrieveDependencyListFromPom(
+    finalDependenciesList: List<MavenListDependency>
+  ): MavenDependencyList {
+    val mavenDependencyList = mutableListOf<MavenDependency>()
+    finalDependenciesList.forEach {
+      // Remove ".jar" or ".aar" or any other extension from the specified url.
+      val pomFileUrl = "${it.url?.dropLast(3)}pom"
+      val artifactName = it.coord
+      val artifactVersion = StringBuilder()
+      var lastIndex = artifactName.length - 1
+      while (lastIndex >= 0 && artifactName[lastIndex] != ':') {
+        artifactVersion.append(artifactName[lastIndex])
+        lastIndex--
+      }
+      artifactVersion.reverse()
+      val pomFile = licenseFetcher.scrapeText(pomFileUrl)
+      val mavenDependency = MavenDependency.newBuilder().apply {
+        this.artifactName = it.coord
+        this.artifactVersion = artifactVersion.toString()
+        this.addAllLicense(extractLicenseLinksFromPom(pomFile))
+      }
+      mavenDependencyList.add(mavenDependency.build())
+    }
+    return MavenDependencyList.newBuilder().addAllMavenDependency(mavenDependencyList).build()
+  }
+
+  private fun retrieveThirdPartyMavenDependenciesList(
+    rootPath: String
+  ): List<String> {
+    return BazelClient(File(rootPath).absoluteFile, commandExecutor)
+      .retrieveThirdPartyMavenDepsListForBinary("oppia")
+      .map { dep ->
+        if (dep.startsWith(MAVEN_PREFIX)) dep.substring(MAVEN_PREFIX.length, dep.length) else dep
```

Comment:
```suggestion
        if (dep.startsWith(MAVEN_PREFIX)) dep.substring(MAVEN_PREFIX.length) else dep
```

substring automatically goes to the end of the string if no endIndex is specified.

That being said, I suggest using ``removePrefix`` instead: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/remove-prefix.html.

Reply:
Done.